src/styles/main.scss

```
@@ -1,3 +1,64 @@
+$dayWidth: 100px;
+$dayInWeek: 7;
+$gap: 1px;
+$calendarWidth: $dayWidth * $dayInWeek + ($dayInWeek - 1) * $gap;
+
 body {
   margin: 0;
 }
+
+.calendar {
+  padding: 30px 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: $calendarWidth;
+
+  
+  @for $length from 28 through 31 {
+    &__month-length--#{$length} :nth-child(n + #{$length + 1}) {
+      visibility: hidden;
+    }
+  }
+
+  $day-of-week: 
+    "mon" 1,
+    "tue" 2,
+    "wed" 3,
+    "thu" 4,
+    "fri" 5,
+    "sat" 6,
+    "sun" 7;
+
+  @each $name, $number in $day-of-week {
+    &__start-day--#{$name} :first-child {
+      margin-left: $dayWidth * ($number - 1) + $gap * ($number - 1);
+    }
+  }
+}
+
+.day {
+  box-sizing: border-box;
+  display: flex;
+  justify-content: center;
+  height: 100px;
+  width: $dayWidth;
+  background-color: #eee;
+  border: 1px solid black;
+  transition-duration: 0.5s;
```

Always list styles that you apply transition for:
```suggestion
  transition: background-color 0.5s, transform 0.5s;
```